elfpy/data/acquire_data.py

```diff
@@ -225,6 +240,27 @@ def main(
                         continue
                 if block_pool_info:
                     postgres.add_pool_infos([block_pool_info], session)
+
+                # keep trying until it returns
+                block_checkpoint_info = None
+                for _ in range(RETRY_COUNT):
```

Comment:
just wondering -- why would this break in the first place? no checkpoints yet?

Reply:
I can't speak to getting checkpoints, but generally I found that anything querying the chain using web3 sometimes fails without reason, and retrying typically fixes it

Reply:
this makes me think we should add retries to our generic smart_contract_read_call instead of replicating it everywhere.